### PR TITLE
docs: api/grn_obj: move grn_obj_path_by_id() reference to the header file

### PR DESCRIPTION
**Pull request by otegami**
GitHub: GH-1817

Prepare to switch to documents automatically generated by Doxygen.

### CODE REVIEW

File: include/groonga/groonga.h [hunk 1]
```diff
@@ -1718,6 +1718,19 @@ grn_column_get_all_index_columns(grn_ctx *ctx,
...
+ * \brief Return the pathname of an object ID.
+ *
+ * \param ctx The context object.
+ * \param db The target database object.
+ * \param id The ID of the object whose pathname is to be obtained.
+ * \param buffer A pre-allocated buffer where the resulting pathname will be
+ *               stored.
+ *
+ * \return \ref GRN_SUCCESS on success, or the appropriate \ref grn_rc on error.
+ *         For example, \ref GRN_INVALID_ARGUMENT is returned if db` is not
```
Comment (kou) [inline include/groonga/groonga.h:1731]:
```suggestion
 *         For example, \ref GRN_INVALID_ARGUMENT is returned if `db` is not
```

File: include/groonga/groonga.h [hunk 2]
```diff
@@ -1718,6 +1718,19 @@ grn_column_get_all_index_columns(grn_ctx *ctx,
...
 GRN_API grn_rc
 grn_obj_delete_by_id(grn_ctx *ctx, grn_obj *db, grn_id id, bool remove_p);
+/**
+ * \brief Return the pathname of an object ID.
+ *
+ * \param ctx The context object.
+ * \param db The target database object.
+ * \param id The ID of the object whose pathname is to be obtained.
+ * \param buffer A pre-allocated buffer where the resulting pathname will be
+ *               stored.
```
Comment (kou) [inline include/groonga/groonga.h:1728]:
```suggestion
 *               stored. For example, `char buffer[PATH_MAX]` can be used.
```

File: include/groonga/groonga.h (hunk 1 above)
Comment (otegami) [reply include/groonga/groonga.h]: fix: [e669fb7](https://github.com/groonga/groonga/pull/2021/commits/e669fb7f1aefa9ba768c4f4940823c3008c28b0f)

File: include/groonga/groonga.h (hunk 2 above)
Comment (otegami) [reply include/groonga/groonga.h]: fix: [e669fb7](https://github.com/groonga/groonga/pull/2021/commits/e669fb7f1aefa9ba768c4f4940823c3008c28b0f)

**Comment by otegami** [comment]
Thank you for reviewing it.

File: include/groonga/groonga.h [hunk 3]
```diff
@@ -1718,6 +1718,19 @@ grn_column_get_all_index_columns(grn_ctx *ctx,
...
+/**
+ * \brief Return the pathname of an object ID.
+ *
+ * \param ctx The context object.
+ * \param db The target database object.
+ * \param id The ID of the object whose pathname is to be obtained.
+ * \param buffer A pre-allocated buffer where the resulting pathname will be
+ *               stored. For example, `char buffer[PATH_MAX]` can be used.
+ *
+ * \return \ref GRN_SUCCESS on success, or the appropriate \ref grn_rc on error.
```
Comment (kou) [inline include/groonga/groonga.h:1730]:
```suggestion
 * \return \ref GRN_SUCCESS on success, the appropriate \ref grn_rc on error.
```

**Pull request merged**